erpnext/assets/doctype/asset/asset.js

```diff
@@ -799,15 +799,35 @@ frappe.ui.form.on("Asset Finance Book", {
 });
 
 erpnext.asset.scrap_asset = function (frm) {
-	frappe.confirm(__("Do you really want to scrap this asset?"), function () {
-		frappe.call({
-			args: {
-				asset_name: frm.doc.name,
+	var scrap_dialog = new frappe.ui.Dialog({
+		title: "Enter date to scrap asset",
+		fields: [
+			{
+				label: __("Select the date"),
+				fieldname: "scrap_date",
+				fieldtype: "Date",
+				reqd: 1,
 			},
-			method: "erpnext.assets.doctype.asset.depreciation.scrap_asset",
-			callback: (r) => frm.reload_doc(),
-		});
+		],
+		size: "medium",
+		primary_action_label: "Submit",
+		primary_action(values) {
+			frappe.call({
+				args: {
+					asset_name: frm.doc.name,
+					scrap_date: values.scrap_date,
+					purchase_date: frm.doc.purchase_date,
+					calculate_depreciation: frm.doc.calculate_depreciation,
+				},
+				method: "erpnext.assets.doctype.asset.asset.validate_asset_scrap_date",
```

Comment:
We should call the `scrap_asset` method from here as it is the primary task and `validate_asset_scrap_date` should be called from inside the `scrap_asset function`